### PR TITLE
Add `typos` to pre-commit config

### PR DESCRIPTION
**Pull request by afuetterer**
## Description

Related issue: #1005

## Types of Changes
- [x] Build related changes

## Checklist

- [x] I have read the [contributor guide](https://github.com/rdmorganiser/rdmo/blob/main/CONTRIBUTING.md).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by afuetterer** [comment]
I added the following directories and patterns to the exclude list, meaning `typos` will not check those.

- rdmo/locale/.*
- rdmo/.*_de.html$
- rdmo/.*_es.html$
- rdmo/.*_fr.html$
- rdmo/.*_it.html$
- testing/.*.json # contains german
- testing/.*.xml # contains german

**Comment by afuetterer** [comment]
typos wants to change `toggleDescandants` to `toggleDescendants` in a few js files. Is that alright?

File: rdmo/core/templates/core/home.html [hunk 1]
```diff
@@ -29,7 +29,7 @@
         <div class="header-image header-image-2">
             <img src="{% static 'core/img/header/collection.jpg' %}"></img>
             <p class="attribution">
-                <a href="https://www.flickr.com/photos/bulle_de/9221247815">Nasssammlung (Christopher Bulle)</a> / <a href="https://creativecommons.org/licenses/by/2.0/">CC BY 2.0</a>
+                <a href="https://www.flickr.com/photos/bulle_de/9221247815">Nasssammlung (Christopher Bulle)</a> / <a href="https://creativecommons.org/licenses/by/2.0/">CC BY 2.0</a> <!-- spellchecker:disable-line -->
```
Comment (afuetterer) [inline rdmo/core/templates/core/home.html:32]: I added a comment to the line containing the name "Bulle". Typos wanted to fix it, as Bullet.

**Review by jochenklar — APPROVED** [review]
Thanks!

**Comment by afuetterer** [comment]
> typos wants to change `toggleDescandants` to `toggleDescendants` in a few js files. Is that alright?

I still need a decision here. Should I rename these? CI/typos fails at the moment.

**Comment by jochenklar** [comment]
Sorry, yes, its fine.

**Comment by afuetterer** [comment]
Alright, will rename those.

**Pull request merged**